index.html

```diff
@@ -5099,6 +5125,31 @@ <h2 id="defining-a-valid-audioworkletprcoessor">
                 <code>"name"</code>.
               </li>
             </ol>
+            <p>
+              The return value of this method controls the lifetime of the
+              <a>AudioWorkletProcessor</a>'s associated <a>AudioWorkletNode</a>
+              by determining the existence of the node's <a href=
+              "#dfn-active-reference">active reference</a>. At the conclusion
+              of each call to the <code>process()</code> method, if the result
+              of applying <a href=
+              "https://tc39.github.io/ecma262/#sec-toboolean"><code>ToBoolean</code></a>
+              (described in [[!ECMASCRIPT]]) to the return value is equal to
+              <code>true</code>, then a single <a>active reference</a> is
+              retained to the node, affecting the node's <a href=
+              "#lifetime-AudioNode">lifetime</a>. Otherwise, any active
+              reference is removed and the node's lifetime depends only on
+              other references as they may exist.
+            </p>
+            <div class="note">
+              While returning <code>true</code> ensures that
+              <code>process()</code> will be called during subsequent rendering
+              quanta, returning <code>false</code> does not prevent future
+              calls to <code>process()</code>. Other references to the
+              corresponding <a>AudioWorkletNode</a> may exist and the node may
+              thus continue to function. Furthermore, these future calls may
+              return a value of <code>true</code>, reinstating the active
+              reference.
```

Comment:
@rtoy I've had some time to think about this.

In line with @hoch's suggestion, how about this, then: : if `process()` returns false _and_ there are no connected inputs to the AWN, then `process()` will never be called again and the node is effectively dead. 

We cannot get rid of the "active reference" concept altogether, because the Lifetime section still needs to state that the UA should keep a source-style AWN/AWP in memory that has no JS references but is still producing sound in the graph. What I'm getting rid of here is the flaky behavior and "reinstatement" that you didn't like.

This is slightly different from @hoch's proposal in just one respect: the behavior does not matter whether there is a JS reference or not. I don't see how the engine can know whether a JS reference exists or not, until a GC actually occurs.

But this seems immaterial. If there was a JS reference to a disconnected node that returns false from `process()`, the node can still become dead for all time and its AWP never gets called again. The AWN just becomes useless at that point (sort of like holding onto an ABSN that has stopped).

Thoughts?